src/Microsoft.NetFramework.Analyzers/Core/DoNotUseInsecureDtdProcessing.cs

```diff
@@ -256,15 +256,24 @@ private void AnalyzeMethodOverloads(OperationAnalysisContext context, IMethodSym
 
                     if (xmlReaderSettingsIndex < 0)
                     {
-                        DiagnosticDescriptor rule = RuleDoNotUseInsecureDtdProcessing;
-                        Diagnostic diag = Diagnostic.Create(
-                                RuleDoNotUseInsecureDtdProcessing,
-                                expressionSyntax.GetLocation(),
-                                SecurityDiagnosticHelpers.GetLocalizableResourceString(
-                                    nameof(MicrosoftNetFrameworkAnalyzersResources.XmlReaderCreateWrongOverloadMessage)
-                                )
-                            );
-                        context.ReportDiagnostic(diag);
+                        if (method.Parameters.Length == 1 
+                            && method.Parameters[0].RefKind == RefKind.None
+                            && method.Parameters[0].Type.SpecialType == SpecialType.System_String)
+                        {
+                            // inputUri can load be a URL.  Should further investigate if this is worth flagging.
```

Comment:
Our docs documentation has a sample that flags this. See second violation here: https://docs.microsoft.com/en-us/visualstudio/code-quality/ca3075-insecure-dtd-processing?view=vs-2017#violations-1

```
using System.Xml;

namespace TestNamespace
{
    class TestClass
    {
        private static void TestMethod()
        {
            var reader = XmlTextReader.Create(""doc.xml""); //warn
        }
    }
}
```

Reply:
Right, but I'm not sure what it's supposed to vulnerable to.

Reply:
Sure, make sure you file a doc bug if you change the semantics here :-)